presto-main/src/main/java/io/prestosql/sql/planner/QueryPlanner.java

```diff
@@ -161,6 +185,172 @@ public RelationPlan plan(Query query)
                 outerContext);
     }
 
+    public RelationPlan planExpand(Query query)
+    {
+        checkArgument(analysis.isExpandableQuery(query), "query is not registered as expandable");
+
+        Union union = (Union) query.getQueryBody();
+        ImmutableList.Builder<PlanNode> recursionSteps = ImmutableList.builder();
+
+        // plan anchor relation
+        Relation anchorNode = union.getRelations().get(0);
+        RelationPlan anchorPlan = new RelationPlanner(analysis, symbolAllocator, idAllocator, lambdaDeclarationToSymbolMap, metadata, outerContext, session, replacementMap)
+                .process(anchorNode, null);
+
+        // prune anchor plan outputs to contain only the symbols exposed in the scope
+        List<Symbol> anchorOutputs = anchorPlan.getRoot().getOutputSymbols().stream()
+                .filter(ImmutableSet.copyOf(anchorPlan.getFieldMappings())::contains)
+                .collect(toImmutableList());
+        PlanNode anchorPlanRoot = new ProjectNode(idAllocator.getNextId(), anchorPlan.getRoot(), Assignments.identity(anchorOutputs));
```

Comment:
This is necessary for successful replacement. At some point we must plug recursion step plan in the place where anchor plan was.
As both anchor plan and step plan might have additional output symbols, they need to be pruned so that the replacement fits and the output symbols match their "continuations".